src/track.rs

```diff
@@ -0,0 +1,37 @@
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize, PartialEq, Eq)]
+pub struct Track {
+    pub album: String,
+    pub artists: Vec<String>,
+    #[serde(with = "humantime_serde")]
+    pub duration: std::time::Duration,
+    pub explicit: bool,
+    pub href: Option<String>,
+    pub id: Option<rspotify::model::TrackId<'static>>,
+    pub name: String,
+    pub popularity: u32,
+}
+
+impl From<&rspotify::model::FullTrack> for Track {
+    fn from(track: &rspotify::model::FullTrack) -> Self {
+        let album = track.album.name.clone();
+        let artists = track
+            .artists
+            .iter()
+            .map(|artist| artist.name.clone())
+            .collect::<Vec<String>>();
+        let duration =
+            std::time::Duration::from_secs(track.duration.num_seconds().try_into().unwrap());
```

Comment:
Is this unwrap can't panic? If it can, perhaps it's better to use TryFrom?

Reply:
Totally agree, I will change this now.